docroot/profiles/custom/cgov_site/modules/custom/pdq_cancer_information_summary/src/OrphanCleanup.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Drupal\pdq_cancer_information_summary;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Database\Connection;
+
+/**
+ * Service for clearing out orphan summary sections.
+ *
+ * @package Drupal\pdq_cancer_information_summary
+ */
+class OrphanCleanup {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The database connection used to find orphaned summary sections.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $connection;
+
+  /**
+   * Constructs a RevisionPruner object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Database\Connection $connection
+   *   The database connection which will be used to find orphans.
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entity_type_manager,
+    Connection $connection
+  ) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->connection = $connection;
+  }
+
+  /**
+   * Drop orphaned PDQ summary sections.
+   *
+   * Identify `paragraph` entities of type `pdq_summary_section` for which at
+   * least some of the revisions no longer have (or never had) a parent node.
+   * If all of such an entity's revisions are orphans, delete the entire
+   * entity. Otherwise, delete only revisions which are orphaned. Perform as
+   * many complete entity deletions as are requested by the caller, if that
+   * many are eligible for deletion, counting each entity as a single deletion,
+   * regardless of how many revisions the entity has. If there are fewer
+   * entities eligible for deletion than the number of deletions requested,
+   * perform the remaining number of deletions on individual orphaned revisions
+   * from `paragraph` entities which have some non-orphaned revisions (or as
+   * many such revisions as are eligible for deletion). It is not an error to
+   * perform fewer deletions than requested. If no maximum number of deletions
+   * is specified by the caller, delete all orphans found.
+   *
+   * @param mixed $max_deletions
```

Comment:
My thought was to be able to distinguish between the case when the caller passed in a value and when it didn't (hence the default value of `FALSE`). However, since I decided not to use the value if the caller passed 0 that distinction doesn't serve any real purpose, and we can tighten up the parameter type as you suggest. Changed the default to 0 and the test to `if ($max_deletions > 0)` (and the parameter comment type to `int`).